x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/NodesCachesStatsIntegTests.java

```diff
@@ -0,0 +1,161 @@
+        createIndex(index, Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build());
+
+        final int nbDocs = randomIntBetween(1_000, 10_000);
+        try (BackgroundIndexer indexer = new BackgroundIndexer(index, "_doc", client(), nbDocs)) {
+            waitForDocs(nbDocs, indexer);
+        }
+        refresh(index);
+
+        final String repository = "repository";
+        createRepository(repository, FsRepository.TYPE);
+
+        final String snapshot = "snapshot";
+        createFullSnapshot(repository, snapshot);
+
+        assertAcked(client().admin().indices().prepareDelete(index));
+
+        final String mountedIndex = "mounted-index";
+        mountSnapshot(repository, snapshot, index, mountedIndex, Settings.EMPTY, Storage.SHARED_CACHE);
+        ensureYellowAndNoInitializingShards(mountedIndex);
+
+        assertExecutorIsIdle(SearchableSnapshotsConstants.CACHE_FETCH_ASYNC_THREAD_POOL_NAME);
+
+        final NodesCachesStatsResponse cachesStatsResponse = client().execute(
+            TransportSearchableSnapshotsNodeCachesStatsAction.TYPE,
+            new NodesRequest(Strings.EMPTY_ARRAY)
+        ).actionGet();
+        assertThat(cachesStatsResponse.getNodes(), hasSize(internalCluster().numDataNodes()));
+        assertThat(cachesStatsResponse.hasFailures(), equalTo(false));
+
+        for (NodeCachesStatsResponse nodeCachesStats : cachesStatsResponse.getNodes()) {
+            final String nodeId = nodeCachesStats.getNode().getId();
+            final String nodeName = nodeCachesStats.getNode().getName();
+
+            final ClusterService clusterService = internalCluster().getInstance(ClusterService.class, nodeName);
+            final long totalFsSize = client().admin()
+                .cluster()
+                .prepareNodesStats(nodeId)
+                .clear()
+                .setFs(true)
+                .get()
+                .getNodesMap()
+                .get(nodeId)
+                .getFs()
+                .getTotal()
+                .getTotal()
+                .getBytes();
+
+            final long cacheSize = FrozenCacheService.calculateCacheSize(clusterService.getSettings(), totalFsSize);
+            assertThat(nodeCachesStats.getSize(), equalTo(cacheSize));
+
+            final long regionSize = FrozenCacheService.SNAPSHOT_CACHE_REGION_SIZE_SETTING.get(clusterService.getSettings()).getBytes();
+            assertThat(nodeCachesStats.getRegionSize(), equalTo(regionSize));
+
+            assertThat(nodeCachesStats.getNumRegions(), equalTo(Math.toIntExact(cacheSize / regionSize)));
+            assertThat(nodeCachesStats.getWrites(), equalTo(0L));
+            assertThat(nodeCachesStats.getBytesWritten(), equalTo(0L));
+            assertThat(nodeCachesStats.getReads(), equalTo(0L));
+            assertThat(nodeCachesStats.getBytesRead(), equalTo(0L));
+            assertThat(nodeCachesStats.getEvictions(), equalTo(0L));
```

Comment:
We only assert that evictions is 0. I wonder if we could clear cache after the search and then add an assert that evictions is greater than 0?

Reply:
That's a good suggestion, thanks! I pushed dd84999afa9f2d3664c9254c1e0aa224fff907e6